mods/coop/hook/lua/ScenarioFramework.lua

```diff
@@ -118,7 +134,16 @@ end
 -- @param failureDialogue A VO to play explaining how much of a failure you are (if any).
 -- @param currentObjectives The AssignedObjectives from the map.
 function PlayerDeath(deadCommander, failureDialogue, currentObjectives)
-    if ScenarioInfo.OpEnded then
+    if failureDialogue then
+        FailDialogue = failureDialogue
+    end
+        local CommanderUnits = GetListOfHumanUnits(categories.COMMAND)
+        for _, unit in CommanderUnits do
+            if unit ~= deadCommander then
+                CreateUnitDeathTrigger(PlayerDeath, unit, true)
```

Comment:
Death trigger is already set on every ACU, this will create double trigger, won't it?

Reply:
This is to ensure that the trigger is set on all ACUs.

Reply:
But its gonna lead to double triggers, and then the function being called two times

Reply:
Fixed.